frontend/src/scenes/insights/utils/compareFilters.ts

```diff
@@ -3,7 +3,16 @@ import { objectCleanWithEmpty, objectsEqual } from 'lib/utils'
 
 import { cleanFilters } from './cleanFilters'
 
-const clean = (f: Partial<AnyFilterType>): Partial<AnyFilterType> => objectCleanWithEmpty(cleanFilters(f))
+const clean = (f: Partial<AnyFilterType>): Partial<AnyFilterType> => {
+    const cleanedFilters = objectCleanWithEmpty(cleanFilters(f))
+    cleanedFilters.events = cleanedFilters.events?.map((e) => {
+        if (e.math === 'total') {
+            delete e.math
+        }
```

Comment:
This relates to the "fixes an issue where the queryChanged selector would return true for semantically equal filters" part. Basically we compare filters sent back from the backend with filters generated from the query on the frontend. Some things are functionally equal, but will fail an equality check e.g. a single event can have an `"order": 0` property, but sometimes doesn't. Here the property math can be "total", but sometimes doesn't. Since total is the default that's semantically equal.

Reply:
Ah, alright, let's just add a comment for this so that the logic is documented